staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -88,6 +89,13 @@ func (a statusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Obj
 	if u, ok := obj.(*unstructured.Unstructured); ok {
 		v := obj.GetObjectKind().GroupVersionKind().Version
 
+		// ratcheting validation of x-kubernetes-list-type value map and set
+		if uOld, oldUnstructured := old.(*unstructured.Unstructured); oldUnstructured {
+			if oldErrs := structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], uOld.Object); len(oldErrs) == 0 {
+				errs = append(errs, structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], u.Object)...)
+			}
+		}
+
```

Comment:
@liggitt Could you please review the changes
- Do we need changes inside the function `ValidateListSetsAndMaps` too ? I am asking this because of [this](https://github.com/kubernetes/kubernetes/issues/106401#issuecomment-988043081) comment of yours, are you suggesting changes in these points or are you informing what the suggested changes (in the comment) will do in the program ?